### PR TITLE
Configurable Automatic Refresh Timer

### PR DESCRIPTION
**Pull request by ChristianAtDell**
# Description
Following changes made in PR #1015 there is an intermittent issue where initial updates are missed by the Watchers in the edge case where driver pods crash in the first few seconds of being booted (and thus, the CSM object remains in a Failed state until the next automated refresh). While mutexing has been investigated as a better long-term solution, it has proven to be a challenging point, so the option of allowing a reduced delay between automated refreshes has been chosen to get the object back into a Successful state in a reasonable timeframe. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1910 |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility
- [X] I have executed the relevant end-to-end test scenarios

# How Has This Been Tested?
PowerMax driver (the one that was running into crashing issues that caused this edge case) was reinstalled 300+ times. The object did not get stuck in Failed state for longer than the allowed window of 10 minutes.

### CODE REVIEW

**Review by alexemc — APPROVED** [review]

**Review by anathoodell — APPROVED** [review]

File: controllers/csm_controller.go [hunk 1]
```diff
@@ -538,14 +541,28 @@ func (r *ContainerStorageModuleReconciler) handleDaemonsetUpdate(oldObj interfac
...
-	sharedInformerFactory := sinformer.NewSharedInformerFactoryWithOptions(r.K8sClient, time.Duration(time.Hour))
+	_, log := logger.GetNewContextWithLogger("ContentWatch")
+	refreshMinutes, err := operatorutils.GetEnvironmentVariable(RefreshEnvVar)
+	if err != nil {
+		log.Info("Refresh time environment variable not set, defaulting to 60 minutes")
+		refreshMinutes = "60"
+	}
+	refreshMinutesInt, err := strconv.Atoi(refreshMinutes)
+	if err != nil {
+		log.Info("Refresh time environment variable not a valid number, defaulting to 60 minutes")
```
Comment (alexemc) [inline controllers/csm_controller.go:552]: I'd say it's an Error, not Info.

File: controllers/csm_controller.go [hunk 2]
```diff
@@ -96,6 +96,9 @@ const (
 
 	// CSMVersion -
 	CSMVersion = "v1.15.0"
+
+	// RefreshEnvVar - environment variable name for watcher timed refreshes
+	RefreshEnvVar = "refreshIntervalMinutes"
```
Comment (alexemc) [inline controllers/csm_controller.go:101]: An env variable name is conventionally in uppercase with underscores.

File: controllers/csm_controller.go (hunk 1 above)
Comment (ChristianAtDell) [reply controllers/csm_controller.go:552]: I disagree. It's perfectly fine for the user to not have defined this item; we include it by default in our operator.yaml but it is an optional configuration item with a default value.

**Review by alexemc — DISMISSED** [review]

**Review by kumarkgosa — APPROVED** [review]

**Review by alexemc — APPROVED** [review]
